app/queries/dependent_expansion_rules.rb

```diff
@@ -11,16 +11,27 @@ def expand_field(web_content_item)
       web_content_item.to_h.slice(*expansion_fields(web_content_item.document_type.to_sym))
     end
 
-    def recurse?(link_type)
-      recursive_link_types.include?(link_type.to_sym)
+    def recurse?(link_type, level_index = 0)
+      recursive_link_types.any? do |t|
+        t[level_index] == link_type.to_sym || t.last == link_type.to_sym
+      end
     end
 
     def reverse_name_for(link_type)
       reverse_names[link_type.to_sym]
     end
 
     def recursive_link_types
-      reverse_names.keys - [:documents, :working_groups]
+      [
```

Comment:
As the rules this follows is somewhat opaque do you think it would benefit from a comment?